docs/tutorials/sinks/2_http_sink.md

```diff
@@ -0,0 +1,504 @@
+    client: HttpClient,
+}
+
+impl BasicSink {
+    pub fn new(config: &BasicConfig) -> Self {
+        let tls = TlsSettings::from_options(&None).unwrap();
+        let client = HttpClient::new(tls, &Default::default()).unwrap();
+        let endpoint = config.endpoint.clone();
+
+        Self { client, endpoint }
+    }
+}
+```
+
+# Encoder
+
+Now we want to create an Encoder that will take our event and convert it 
+to raw bytes.
+
+```rust
+#[derive(Clone)]
+struct BasicEncoder;
+```
+
+The Encoder must implement the `Encoder` trait:
+
+```rust
+impl Encoder<Event> for BasicEncoder {
+    fn encode_input(
+        &self,
+        input: Event,
+        writer: &mut dyn std::io::Write,
+    ) -> std::io::Result<usize> {
+    }
+}
+```
+
+The `Encoder` trait is generic over the type of input that we are 
+expecting. In our case it is `Event` since we will be encoding a single
+event at a time. Other Sinks may encode a `Vec<Event>`  if they are 
+sending batches of events, or they may send a completely different type
+if each event is processed in someway prior to encoding.
+
+
+The `encode_input` serializes the event to a String and writes these
+bytes:
+ 
+```rust
+    fn encode_input(
+        &self,
+        input: Event,
+        writer: &mut dyn std::io::Write,
+    ) -> std::io::Result<usize> {
+        let event = serde_json::to_string(&input).unwrap();
+        write_all(writer, 1, event.as_bytes()).map(|()| event.len())
+    }
+```
+
+# Request Builder
+
+Next we creata request builder that turns the event into a request. The 
+request is used by a Tower service that is responsible for actually sending
+the data.
+
+The request looks like:
+
+```rust
+#[derive(Clone)]
+struct BasicRequest {
+    payload: Bytes,
+    finalizers: EventFinalizers,
+    metadata: RequestMetadata,
+}
+```
+
+The fields in the request are:
+
+## payload
+
+The payload is the actual bytes that we will be sending out. These are the bytes
+generated by our `BasicEncoder`.
+
+## finalizers
+
+`EventFinalizers` is a collection of `Finalizers`. A `Finalizer` is used to track 
+the status of a given event and is used to support 
+[end to end acknowledgements](https://vector.dev/docs/about/under-the-hood/guarantees/#acknowledgement-guarantees).
+
+## metadata
+
+The metadata contains additional data that is used to emit various metrics when 
+a request is successfully sent.
+
+
+We need to implement a number of traits for the request to access these fields:
+
+```rust
+impl MetaDescriptive for BasicRequest {
+    fn get_metadata(&self) -> RequestMetadata {
+        self.metadata
+    }
+}
+
+impl Finalizable for BasicRequest {
+    fn take_finalizers(&mut self) -> EventFinalizers {
+        self.finalizers.take_finalizers()
+    }
+}
+```
+
+The request builder must implement the `RequestBuilder<>` trait: 
+
+```rust
+impl RequestBuilder<Event> for BasicRequestBuilder {
+```
+
+There are a number of stages in the request builder process -  first the input 
```

Comment:
perhaps these stages could be listed out? I think seeing lists whether numbered or not, helps break up tutorials and add clarity to steps.